src/client/Client.cc

```diff
@@ -11342,44 +11340,55 @@ int Client::ll_getxattr(Inode *in, const char *name, void *value,
 int Client::_listxattr(Inode *in, char *name, size_t size,
 		       const UserPerm& perms)
 {
+  bool len_only = (size == 0);
+  size_t this_len;
   int r = _getattr(in, CEPH_STAT_CAP_XATTR, perms, in->xattr_version == 0);
-  if (r == 0) {
-    for (map<string,bufferptr>::iterator p = in->xattrs.begin();
-	 p != in->xattrs.end();
-	 ++p)
-      r += p->first.length() + 1;
+  if (r != 0) {
+    goto out;
+  }
 
-    const VXattr *vxattrs = _get_vxattrs(in);
-    r += _vxattrs_name_size(vxattrs);
+  r = 0;
+  for (map<string,bufferptr>::iterator p = in->xattrs.begin();
+       p != in->xattrs.end();
+       ++p) {
```

Comment:
As a rule in new code, when changing or creating new loops like this, we use the generic structured for loop. i.e.:

`for (const auto& p : in->xattrs)`